keras/layers/containers.py

```diff
@@ -239,44 +246,59 @@ def add_input(self, name, input_shape, dtype='float'):
                                   'input_shape': input_shape,
                                   'dtype': dtype})
 
-    def add_node(self, layer, name, input=None, inputs=[],
-                 merge_mode='concat', concat_axis=-1, dot_axes=-1, create_output=False):
+    def add_node(self, layer, name, input=None, merge_mode='concat', concat_axis=-1, create_output=False):
         if hasattr(layer, 'set_name'):
             layer.set_name(name)
+            
         if name in self.namespace:
             raise Exception('Duplicate node identifier: ' + name)
-        if input:
+            
+        if isinstance(input, str):
             if input not in self.namespace:
                 raise Exception('Unknown node/input identifier: ' + input)
-            if input in self.nodes:
-                layer.set_previous(self.nodes[input])
-            elif input in self.inputs:
-                layer.set_previous(self.inputs[input])
-        if inputs:
+            prev = self.nodes[input] if input in self.nodes else self.inputs[input]
+            layer.set_previous(prev)
+        elif isinstance(input, list):
             to_merge = []
-            for n in inputs:
+            for n in input:
                 if n in self.nodes:
                     to_merge.append(self.nodes[n])
                 elif n in self.inputs:
                     to_merge.append(self.inputs[n])
                 else:
-                    raise Exception('Unknown identifier: ' + n)
-            merge = Merge(to_merge, mode=merge_mode, concat_axis=concat_axis, dot_axes=dot_axes)
+                    raise Exception('Unknown identifier: ' + n)                    
```

Comment:
Unnecessary whitespace. Please install a PEP8 linter to spot these issues during development.